tests/utils.ts

```diff
@@ -13,28 +13,54 @@ import _ from "lodash";
 dotenv.config({ path: path.resolve(__dirname, "..", ".env.test") });
 
 const CHAIN_ENDPOINT = requireEnvVar("CHAIN_ENDPOINT");
-
-console.log("CHAIN_ENDPOINT", CHAIN_ENDPOINT);
+const CHAIN_ID = parseInt(process.env.CHAIN_ID || 11155111);
+console.log("CHAIN_ENDPOINT", CHAIN_ENDPOINT, "CHAIN_ID", CHAIN_ID);
 
 // Get wallet address from private key
 export async function getAddress(privateKey: string): Promise<string> {
   const wallet = new ethers.Wallet(privateKey);
   return wallet.address;
 }
 
+interface KeyRequest {
+  chainId: number;
+  address: string;
+  expiredAt: Date;
+  issuedAt: Date;
+  signature: string;
+}
+
 // Generate a signed message from a private key
 export async function generateSignature(
   privateKey: string,
-  expiredAt: number
-): Promise<string> {
+  expiredAtEpoch: number
+): Promise<KeyRequest> {
+  const chainId = CHAIN_ID;
   const wallet = new ethers.Wallet(privateKey);
-  const message = getKeyRequestMessage(wallet.address, expiredAt);
+  const issuedAt = new Date();
+  const expiredAt = new Date(expiredAtEpoch * 1000);
+  const message = getKeyRequestMessage(
+      chainId,
+      wallet.address,
+      issuedAt,
+      expiredAt);
 
-  // console.log("Signing message:", message, "Expired at:", expiredAt);
+  const signature = await wallet.signMessage(message);https://www.prisma.io/docs/orm/prisma-migrate/understanding-prisma-migrate/migration-histories#committing-the-migration-history-to-source-control
 
-  const signature = await wallet.signMessage(message);
+  return { chainId, address: wallet.address, issuedAt, expiredAt, signature };
+}
 
-  return signature;
+// Helper function to generate api key message
+export async function generateAuthPayloadWithApiKey(
```

Comment:
This function doesn’t need the `async` keyword.

Reply:
Is this func used only by the `authWithAPIKey`? I think it can be merged into the `authWithAPIKey`

Reply:
Right, sorry I thought it was in source code. Then this is fine.